framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -423,6 +420,59 @@ private void checkPolymorphicClass(ClassTree classTree) {
         }
     }
 
+    class PolyTypeScanner extends AnnotatedTypeScanner<Boolean, Void> {
+        @Override
+        public Boolean visitDeclared(AnnotatedDeclaredType type, Void aVoid) {
```

Comment:
Extend SimpleAnnotatedTypeScanner instead. Then just override defaultAction.  Default action is called recursively for all parts of the AnnotatedTypeMirror. So you just need to check for poly annotations directly in the type.   You’ll also have to override reduce. See TypeVisitor#checkContainsSameToString.